### PR TITLE
Migrate scheduled parts of Support Tool into Job Processor

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
Currently, we can not have more than one instance of the Support Tool running, meaning that we are not able to do rolling upgrades, so there are outages when we deploy new versions. Also, the Support Tool API can be adversely affected by the workload of sample load and validation, which can affect UI responsiveness.

# What has changed
Migrated the scheduled parts of Support Tool into the Job Processor.

# How to test?
Zero regression - ATs should continue to pass as normal.

# Links
Trello: https://trello.com/c/nOrXHmfn

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+coverage:
+  status:
+    project:
+      default:
+        target: 0%
```
Comment (LukeLoze1) [inline .codecov.yml:5]: Now this is production aimed, can it have a target; even if we say 50% (or less) for now and stick in a card to increase coverage?

File: src/main/java/uk/gov/ons/ssdc/jobprocessor/config/AppConfig.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  @Bean
+  public TaskScheduler taskScheduler() {
+    ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+    taskScheduler.setPoolSize(schedulingPoolSize);
+    return taskScheduler;
+  }
+
+  @PostConstruct
+  public void init() {
+    LoggingConfigs.setCurrent(LoggingConfigs.getCurrent().useJson());
```
Comment (LukeLoze1) [inline src/main/java/uk/gov/ons/ssdc/jobprocessor/config/AppConfig.java:48]: Add:
  @Value("$logging.profile")
  private String loggingProfile;

And change to 

  @PostConstruct
  public void init() {
    if (loggingProfile.equals("STRUCTURED")) {
      LoggingConfigs.setCurrent(LoggingConfigs.getCurrent().useJson());
    }

    TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
  }

File: src/main/java/uk/gov/ons/ssdc/jobprocessor/schedule/RowChunkValidator.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  public boolean processChunk(Job job) {
+    boolean hadErrors = false;
+
+    JobTypeSettings jobTypeSettings =
+        jobTypeHelper.getJobTypeSettings(job.getJobType(), job.getCollectionExercise());
+
+    List<JobRow> jobRows =
+        jobRowRepository.findTop500ByJobAndJobRowStatus(job, JobRowStatus.STAGED);
+
+    boolean getValidationRulesPerRow = false;
```
Comment (LukeLoze1) [inline src/main/java/uk/gov/ons/ssdc/jobprocessor/schedule/RowChunkValidator.java:42]: Just an idea to tidy up this complex area - maybe not for now as this works, but if we stuck a boolean getValidationRulesPerRow in JobTypeHelper for each Type.  

Then here we could just have

boolean getValidationRulesPerRow = jobTypeHelper.isGetValidationPerRow();
ColumnValidator[] columnValidators =  jobTypeSettings.getColumnValidators();  //If this is null, nothing lost

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ssdc/jobprocessor/schedule/RowChunkValidator.java:42]: I had some thoughts about more major refactoring... but decided that it'd be better on another card

File: src/main/java/uk/gov/ons/ssdc/jobprocessor/schedule/RowChunkValidator.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      columnValidators = jobTypeSettings.getColumnValidators();
+    }
+
+    for (JobRow jobRow : jobRows) {
+      JobRowStatus rowStatus = JobRowStatus.VALIDATED_OK;
+      List<String> rowValidationErrors = new LinkedList<>();
+
+      if (getValidationRulesPerRow) {
+        // If it's a sensitive data update, and the row is empty then it means "blank out the data"
+        if (job.getJobType() == JobType.BULK_UPDATE_SAMPLE_SENSITIVE
```
Comment (LukeLoze1) [inline src/main/java/uk/gov/ons/ssdc/jobprocessor/schedule/RowChunkValidator.java:58]: Again maybe not for now but could this be built into the jobTypeSettings functionality
jobTypeSettings.getColumnValidatorForSampleOrSensitiveDataRows(dataRow);

If it's sensitive and Blank then return "new ColumnValidator[0]"

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  - docker network create ssdcrmdockerdev_default
+  - echo "$DOCKER_GCP_PASSWORD" > ~/google-service-account-key.json
+  - export GOOGLE_APPLICATION_CREDENTIALS=~/google-service-account-key.json
+
+install: echo "SKIPPING DEFAULT INSTALL" # Travis has a default maven install. Don't build twice!
+
+script: travis_wait mvn verify jacoco:report
+
+after_success:
+  - bash <(curl -s https://codecov.io/bash)
```
Comment (AdamHawtin) [inline .travis.yml:26]: Should use the new codecov uploader plus integrity checks.

**Review by AdamHawtin — DISMISSED** [review]
Code looks good and passes AT's locally, just the codecov uploader should be updated. Also the new repo needs a licence file.

File: src/main/java/uk/gov/ons/ssdc/jobprocessor/schedule/RowChunkValidator.java (hunk 4 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ssdc/jobprocessor/schedule/RowChunkValidator.java:58]: I had some thoughts about more major refactoring... but decided that it'd be better on another card

**Review by LukeLoze1 — APPROVED** [review]
ATs pass for me, good work

**Review by hbrace — APPROVED** [review]

**Pull request merged**